nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -90,4 +123,177 @@ private static void invalidInstantiationError(
         errorBuilder.createErrorDescription(
             errorMessage, analysis.buildDescription(tree), state, null));
   }
+
+  @SuppressWarnings("UnusedVariable")
+  public static void checkInstantiationForAssignments(
+      Tree tree, Config config, VisitorState state, NullAway analysis) {
+    Tree lhsTree;
+    Tree rhsTree;
+    if (tree instanceof VariableTree) {
+      VariableTree varTree = (VariableTree) tree;
+      lhsTree = varTree.getType();
+      rhsTree = varTree.getInitializer();
+    } else {
+      AssignmentTree assignmentTree = (AssignmentTree) tree;
+      lhsTree = assignmentTree.getVariable();
+      rhsTree = assignmentTree.getExpression();
+    }
+    if (rhsTree == null) {
+      // Possible for VariableTrees with no initializer
+      return;
+    }
+    if (rhsTree instanceof NewClassTree
+        && ((NewClassTree) rhsTree).getIdentifier() instanceof ParameterizedTypeTree) {
+      ParameterizedTypeTreeNullableArgIndices typeWrapper =
+          new ParameterizedTypeTreeNullableArgIndices();
+      typeWrapper.checkAssignmentTypeMatch(
+          tree,
+          ASTHelpers.getType(lhsTree),
+          (ParameterizedTypeTree) ((NewClassTree) rhsTree).getIdentifier(),
+          config,
+          state,
+          analysis);
+    } else {
+      NormalTypeTreeNullableTypeArgIndices typeWrapper = new NormalTypeTreeNullableTypeArgIndices();
+      typeWrapper.checkAssignmentTypeMatch(
+          tree, ASTHelpers.getType(lhsTree), ASTHelpers.getType(rhsTree), config, state, analysis);
+    }
+  }
+}
+
+class ParameterizedTypeTreeNullableArgIndices
+    implements AnnotatedTypeWrapper<Type, ParameterizedTypeTree> {
+  @SuppressWarnings("UnusedVariable")
+  @Override
+  public HashSet<Integer> getNullableTypeArgIndices(ParameterizedTypeTree tree, Config config) {
+    List<? extends Tree> typeArguments = tree.getTypeArguments();
+    HashSet<Integer> nullableTypeArgIndices = new HashSet<Integer>();
+    for (int i = 0; i < typeArguments.size(); i++) {
+      if (typeArguments.get(i).getClass().equals(JCTree.JCAnnotatedType.class)) {
+        JCTree.JCAnnotatedType annotatedType = (JCTree.JCAnnotatedType) typeArguments.get(i);
+        for (JCTree.JCAnnotation annotation : annotatedType.getAnnotations()) {
+          Attribute.Compound attribute = annotation.attribute;
+          if (attribute.toString().equals("@org.jspecify.annotations.Nullable")) {
+            nullableTypeArgIndices.add(i);
+            break;
+          }
+        }
+      }
+    }
+    return nullableTypeArgIndices;
+  }
+
+  public static void superTypeMatchingRHSParameterizedTypeTree(
```

Comment:
Rather than making this a static method, have the enclosing object take the relevant `rhs` into the constructor and store the state in a field.  Then, this method can just take maybe the other parameters.  Similar comment for `checkAssignmentTypeMatch`, and same for the methods in `NormalTypeTreeNullableTypeArgIndices`